workitem/workitem_repository.go

```diff
@@ -109,6 +122,61 @@ func (r *GormWorkItemRepository) Save(ctx context.Context, wi app.WorkItem) (*ap
 	if err != nil {
 		return nil, errors.NewBadParameterError("Type", wi.Type)
 	}
+	var order float64
+	if wi.Fields[Previousitem] == nil && wi.Fields[Nextitem] == nil {
+		order, err = strconv.ParseFloat(fmt.Sprintf("%v", res.Fields[SystemOrder]), 64)
+		if err != nil {
+			return nil, errors.NewBadParameterError("data.attributes.order", res.Fields[SystemOrder])
+		}
+	} else if wi.Fields[Previousitem] == nil && wi.Fields[Nextitem] != nil {
+		nextitem := fmt.Sprintf("%v", wi.Fields[Nextitem])
+		next, err := r.LoadFromDB(nextitem)
+		if err != nil {
+			return nil, err
+		}
+		nextorder, err := strconv.ParseFloat(fmt.Sprintf("%v", next.Fields[SystemOrder]), 64)
+		if err != nil {
+			return nil, errors.NewBadParameterError("data.attributes.nextitem", next.Fields[SystemOrder])
+		}
+
+		order = (0 + nextorder) / 2
+
+	} else if wi.Fields[Previousitem] != nil && wi.Fields[Nextitem] == nil {
+		previtem := fmt.Sprintf("%v", wi.Fields[Previousitem])
+		prev, err := r.LoadFromDB(previtem)
+		if err != nil {
+			return nil, err
+		}
+		prevorder, err := strconv.ParseFloat(fmt.Sprintf("%v", prev.Fields[SystemOrder]), 64)
+		if err != nil {
+			return nil, errors.NewBadParameterError("data.attributes.previousitem", prev.Fields[SystemOrder])
+		}
+		order = prevorder + 1000
+
+	} else {
+		previtem := fmt.Sprintf("%v", wi.Fields[Previousitem])
+		prev, err := r.LoadFromDB(previtem)
+		if err != nil {
+			return nil, err
+		}
+		prevorder, err := strconv.ParseFloat(fmt.Sprintf("%v", prev.Fields[SystemOrder]), 64)
+		if err != nil {
+			return nil, errors.NewBadParameterError("data.attributes.previousitem", prev.Fields[SystemOrder])
+		}
+
+		nextitem := fmt.Sprintf("%v", wi.Fields[Nextitem])
+		next, err := r.LoadFromDB(nextitem)
+		if err != nil {
+			return nil, err
+		}
+		nextorder, err := strconv.ParseFloat(fmt.Sprintf("%v", next.Fields[SystemOrder]), 64)
+		if err != nil {
+			return nil, errors.NewBadParameterError("data.attributes.nextitem", next.Fields[SystemOrder])
+		}
+
+		order = (prevorder + nextorder) / 2
```

Comment:
Yes @kwk . I think it should work just fine anyway, but we need to ensure that in case there is a precision overflow , we need to handle the error well.